.github/workflows/mirror_repo.sh

```
@@ -0,0 +1,35 @@
+# If you trigger an update more frequently, it may not start immediately.
+# Make sure sync is finished after sync request was submitted
+MAX_RETRIES=15
+RETRY_INTERVAL=30
+# Convert timestamps to epoch for comparison, timezone agnostic
+SYNC_TIME_SECONDS=$(date -d "$SYNC_TIME" +%s)
+for i in $(seq 1 $MAX_RETRIES); do
+    MIRROR_INFO=$(curl --silent --header "PRIVATE-TOKEN: ${GITLAB_ACCESS_TOKEN}" "${GITLAB_MIRROR_URL}")
+    MIRROR_STATUS=$(echo "$MIRROR_INFO" | jq -r '.update_status')
+    LAST_UPDATE=$(echo "$MIRROR_INFO" | jq -r '.last_update_at')
+    LAST_UPDATE_SECONDS=$(date -d "$LAST_UPDATE" +%s)
+    if [ "$MIRROR_STATUS" = "finished" ] && [ $LAST_UPDATE_SECONDS -gt $SYNC_TIME_SECONDS ]; then
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)